osquery/tables/system/posix/xattr_utils.cpp

```diff
@@ -0,0 +1,189 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifndef __FreeBSD__
+
+#include <string>
+#include <vector>
+
+#include <sys/xattr.h>
+
+#include <osquery/logger.h>
+
+#include "osquery/tables/system/posix/xattr_utils.h"
+
+#ifdef __APPLE__
+#include <TargetConditionals.h>
+
+#ifndef TARGET_OS_MAC
+#error Unsupported macOS target
+#endif
+
+#include "osquery/tables/system/darwin/special_xattr_decoder.h"
+
+#else
+#include "osquery/tables/system/linux/special_xattr_decoder.h"
+#endif
+
+namespace osquery {
+namespace {
+// Compatibility wrappers for macOS
+#if defined(TARGET_OS_MAC)
```

Comment:
How is this different that __APPLE__

Reply:
The APPLE define is enabled for any Apple device (like iOS or iPad). To make sure that it's macOS with a standard filesystem I check for that define

Reply:
We don't support *OS (minus macOS) so I don't think we need this check